frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -52,11 +52,14 @@ const updateConfigMap = async (configMap: ConfigMapKind, key: string, value: str
     ];
     try {
       await k8sPatch(ConfigMapModel, configMap, patch);
+      return Promise.resolve();
     } catch (err) {
       // eslint-disable-next-line no-console
       console.error(err);
+      return Promise.reject();
     }
   }
+  return Promise.resolve();
```

Comment:
i don't think we need to return here as well as we are already returning in try catch block and this is an async function so we only need to return the value or throw an error. There's no need to return a Promise

Reply:
the try catch is inside If block. So we need to return outside of the if block as well

Reply:
ah i should have had this condition out of `updateConfigMap` and in caller, my bad and can still avoid return explicit Promise